File: JAVA/Paciente.java

public class Paciente  extends pessoa implements Cadastro{
	 private int registro;
	 private String data;
	 private String periodo;
	 private String nomeMae;
	 private String examesSolicitados;
	 
	 public Paciente() {
		super();
	 }
	 
	 public Paciente(int registro, String nomeCompleto, String dataNascimento, String telefone, String email,
                    String data, String periodo, String nomeMae, String examesSolicitados) {
        super(nomeCompleto, dataNascimento, telefone, email);
        this.registro = registro;
        this.data = data;
        this.periodo = periodo;
        this.nomeMae = nomeMae;
        this.examesSolicitados = examesSolicitados;
    }

	public int getRegistro() {
		return registro;
	}

	public void setRegistro(int registro) {
		this.registro = registro;
	}

	public String getData() {
		return data;
	}

	public void setData(String data) {
		this.data = data;
	}

	public String getPeriodo() {
		return periodo;
	}

	public void setPeriodo(String periodo) {
		this.periodo = periodo;
	}

	public String getNomeMae() {
		return nomeMae;
	}

	public void setNomeMae(String nomeMae) {
		this.nomeMae = nomeMae;
	}

	public String getExamesSolicitados() {
		return examesSolicitados;
	}

	public void setExamesSolicitados(String examesSolicitados) {
		this.examesSolicitados = examesSolicitados;
	}

 @Override
    public void cadastrar() {
        System.out.println("Paciente cadastrado.");
    }

    @Override
    public void editar() {
        System.out.println("Paciente editado.");
    }

    @Override
    public void exibir() {
        System.out.println(this.toString());
    }

	@Override
	public String toString() {
		return super.toString() + "Paciente: Registro:" + registro + "Data:" + data + "Periodo:" + periodo + "Nome mãe:" + nomeMae
				+ "Exames Solicitados:" + examesSolicitados + "\n";
	}
	 
	 

}

File: Node/paciente.js
const express = require('express');
const router = express.Router();
const Paciente = require('../models/Paciente');

let pacientes = [];
let nextRegistro = 1;

// Listar todos
router.get('/', (req, res) => {
  res.json(pacientes);
});

// Buscar por registro
router.get('/:registro', (req, res) => {
  const registro = parseInt(req.params.registro);
  const paciente = pacientes.find( p => p.registro === registro);
  if (!paciente) return res.status(404).json({ message: 'paciente não encontrado' });
  res.json(paciente);
});

// Inserir
router.post('/', (req, res) => {
  const {nome, telefone, registro, data, periodo, nomeMae, examesSolicitados, Email, Data_Nascimento} = req.body;
  const paciente = new Paciente(nextRegistro++, nome, telefone, registro, data, periodo, nomeMae, examesSolicitados, Email, Data_Nascimento);
  pacientes.push(paciente);
  res.status(201).json(paciente);
});

// Editar
router.put('/:registro', (req, res) => {
  const registro = parseInt(req.params.registro);
  const paciente = pacientes.find(p => p.registro === registro);
  if (!paciente) return res.status(404).json({ message: 'paciente não encontrado' });

  const {nome, telefone, registro: novoRegistro, data, periodo, nomeMae, examesSolicitados, Email, Data_Nascimento} = req.body;
  paciente.nome = nome ?? paciente.nome;
  paciente.telefone =  telefone ?? paciente.telefone;
  paciente.registro = novoRegistro ?? paciente.registro;
  paciente.data = data ?? paciente.data;
  paciente.periodo = periodo ?? paciente.periodo;
  paciente.nomeMae = nomeMae ?? paciente.nomeMae;
  paciente.examesSolicitados = examesSolicitados ?? paciente.examesSolicitados;
  paciente.Email = Email ?? paciente.Email;
  paciente.Data_Nascimento = Data_Nascimento ?? paciente.Data_Nascimento;


  res.json(paciente);
});

// Excluir
router.delete('/:registro', (req, res) => {
  const registro = parseInt(req.params.registro);
  const index = pacientes.findIndex(p => p.registro === registro);
  if (index === -1) return res.status(404).json({ message: 'paciente não encontrado' });

  pacientes.splice(index, 1);
  res.status(204).send();
});

module.exports = router;
